lib/Zonemaster/Backend/DB/PostgreSQL.pm

```diff
@@ -295,6 +295,37 @@ sub add_batch_job {
     return $batch_id;
 }
 
+sub process_unifinished_tests {
```

Comment:
There's a lot of duplication between the DB adapters here. Do you think it would be possible to refactor this to put the main logic in DB.pm and only have the DB specific details in the adapters?

Reply:
The only way is to add two "query creation" methods, one for the select and one for the update, to each backend specific module and use them in DP.pm.
Do you prefer to have it done like this?

Reply:
Yeah, I think so. In principle it's a better design make the database adapters have well-defined APIs of small methods in order to keep them as free from business logic as possible. That way we can test the business logic with mocked out database adapters.